src/dev/flang/be/interpreter/Excecutor.java

```diff
@@ -565,10 +569,100 @@ public Object contract(int cl, ContractKind ck, Value cc)
    */
   Value callOnInstance(int cc, Instance cur, Value outer, List<Value> args, boolean pre)
   {
+    FuzionThread.current()._callStackFrames.push(_fuir.clazzForInterpreter(cc));
+    var pos = this.pos();
```

Comment:
@fridis How bad is this considering performance?

Reply:
`Thread.current()` is very cheap, `FUIR.clazzForInterpreter()` is probably slightly more expensive than just pushing `cc`, which will be turned into `Integer.valueOf(cc)`. 
I fear that `this.pos()` might be the most expensive part, maybe add `AbstractInterpreter.site()` that returns `_fuir.siteFromCI(_c,_i)`, that is maybe most future-proof.
We might even make a site unique for each outer clazz in the future, then you should be able to drop the `_callStackFrames` as well and use something like `_fuir.siteOuterClazz(site)` or similar.